packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3372,12 +3386,21 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   ///
   /// The element will still have a valid parent when this is called. After this
   /// is called, [deactivateChild] is called to sever the link to this object.
+  ///
+  /// [update] is responsible for updating or creating the new child that will
+  /// replace this [child].
   @protected
   @mustCallSuper
   void forgetChild(Element child) {
+    // We should not remove global key reservations in this method. The child
```

Comment:
Maybe: "This method is called on the old parent when the given child (with a global key) is given a new parent. We cannot remove the global key reservation directly in this method because the forgotten child is not removed from the tree until this Element is updated in [update]. If [update] is never called, the forgotten child still represents a global key duplication that we need to catch."